src/averageAges.js

```diff
@@ -15,47 +15,72 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(man => man.sex === 'm');
+  const menCentury = century ? men.filter(
+    man => Math.ceil(man.died / 100) === century) : men;
+
+  const averageAge = menCentury.reduce(
+    (sum, man) => sum + (man.died - man.born), 0) / menCentury.length;
```

Comment:
Create a separate function for the average age calculation. You use the same logic several times.